src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
-
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a < 0 || b < 0 || c < 0) {
+      throw new Error('The length of a side must be more than 0');
+    }
+
+    if (a >= (b + c) || b >= (a + c) || c >= (a + b)) {
+      throw new Error('The length of a side is too long');
+    }
+  }
 
+  getArea():number {
+    const perimeter = (this.a + this.b + this.c) / 2;
```

Comment:
That's a half of perimeter
```suggestion
    const halfPerimeter = (this.a + this.b + this.c) / 2;
```